hathor/p2p/entrypoint.py

```diff
@@ -39,10 +42,14 @@ class Entrypoint:
     peer_id: PeerId | None = None
```

Comment:
Should we add a `tcp6` protocol instead of trying to match a regex? Alternatively, we could use the `[host]` pattern for all ipv6 hosts.

Reply:
This is a good question, I thought about this at some point.

I chose not to rely so much in user input for this and just detect whether an IPv6 was provided.

Otherwise, someone could wrongly use `tcp` instead of `tcp6` as an entrypoint, and it would be relayed as if it was an IPv4 to others, that would error out while trying to parse it, or at least keep trying to connect to it and getting timeouts.

If we do start using `tcp6`, I think it would be better to keep the regex anyway for validation purposes to avoid the scenario above.

What do you think?

One of the advantages of using `tcp6` I thought about initially was that it could be used to also inform about DNS names holding AAAA records. For instance, `tcp6://abc.xyz` would be a way for the user to inform that this domain should be treated as IPv6. This something not supported in the current implementation with the Regex.

We would still have the validation issue, though, which is even harder in this case because it needs a DNS query.

Reply:
Got it. I agree with you. The current implementation seems good to me.